### PR TITLE
fix: make kornia-apriltag tests portable on windows

### LINKED ISSUE #805: [Bug]: kornia-apriltag tests are not portable on Windows due to pthread dependency and fixture path assumptions
### 🐛 Describe the bug

cargo test -p kornia-apriltag is not portable across environments:

On Windows, test/dev dependency resolution pulls apriltag-sys and fails due to missing pthread.h.
AprilTag fixture tests use fragile relative paths and fail when run from a different cwd or when fixture directories are not present.
This causes local test failures unrelated to the actual Rust implementation.

### 🔄 Steps to Reproduce

```bash
1. Use Windows (MSVC toolchain).
2. Clone repo and run from workspace root: cargo test -p kornia-apriltag
3. Observe:   
apriltag-sys compile error: fatal error C1083: Cannot open include file: 'pthread.h'
or fixture runtime error: The system cannot find the path specified.
```

### 💻 Minimal Code Example

```rust
// Repro is command-based rather than API-based:
/// cargo test -p kornia-apriltag
```

### ✅ Expected behavior

kornia-apriltag tests should run consistently across supported platforms.
Platform-specific reference dependencies should not break Windows test builds.
Fixture-based tests should resolve paths robustly and/or skip clearly when fixtures are unavailable.

### ❌ Actual behavior

error: failed to run custom build command for `apriltag-sys v0.3.0`
...
fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory
Error: Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." }

### 🔧 Environment

```shell
kornia-rs version: 0.1.11 (workspace)
Rust version (rustc -V): rustc 1.93.0 (254b59607 2026-01-19)
Cargo version (cargo -V): cargo 1.93.0 (083ac5135 2025-12-15)
OS: Windows 11 (x86_64)
Target architecture: x86_64-pc-windows-msvc
Python version (if using Python bindings): N/A
```

### 📝 Additional context

Relevant files:
- crates/kornia-apriltag/Cargo.toml
- crates/kornia-apriltag/src/lib.rs

Potential fix direction:
- Gate `aprilgrid` / `apriltag` dev-dependencies for non-Windows targets.
- Resolve fixture paths using `CARGO_MANIFEST_DIR` and skip with a clear message when fixtures are unavailable.

### 🤝 Contribution Intent

- [x] I plan to submit a PR to fix this bug
- [ ] I'm reporting this bug but not planning to fix it

### PR DESCRIPTION
**Pull request by Sunilkumarchavhan9**
**Fixes/Relates to:** Fixes #805

**Important**:
- Ensure you are assigned to the linked issue before submitting this PR
- This PR should strictly implement what the linked issue describes
- Do not include changes beyond the scope of the linked issue

---

## 🛠️ Changes Made
- [x] Gate `aprilgrid` and `apriltag` dev-dependencies to non-Windows targets.
- [x] Resolve apriltag fixture paths using `CARGO_MANIFEST_DIR` and skip fixture-driven tests when fixtures are unavailable.

---

## 🧪 How Was This Tested?
- [ ] **Unit Tests:** (List new/updated tests)
- [x] **Manual Verification:** Ran `cargo test -p kornia-apriltag --offline` and verified the crate tests pass on Windows after the fix.
- [x] **Performance/Edge Cases:** Verified the test suite no longer fails when fixture directories are missing and that Windows no longer pulls the pthread-dependent reference dev-dependencies.

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [ ] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [x] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] This PR strictly implements what the linked issue describes (no scope creep)
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

---

## 💭 Additional Context
This PR fixes the Windows portability issue reported in #805 by preventing `apriltag-sys` from being pulled into Windows test builds through reference-only dev-dependencies.

It also makes fixture-based apriltag tests more robust by resolving paths from `CARGO_MANIFEST_DIR` and skipping cleanly when fixture directories or matching files are not available in the local checkout.

### CODE REVIEW

**Comment by sidd-27** [comment]
1. Silent skips can mask real failures
Using eprintln! + return Ok(()) when fixtures are missing means CI will pass silently on a misconfigured environment. Consider using #[cfg(not(windows))] on the test functions themselves (consistent with the dep gating), or at minimum emitting a cargo test -- --nocapture-visible warning. A better pattern:
rust#[cfg_attr(windows, ignore = "apriltag fixtures not available on Windows")]
#[test]
fn test_apriltag_detection() { ... }
2. images_dir: &str mixed with Path::new(...).join(images_dir)
The function signature still takes &str but now treats it as a relative path from CARGO_MANIFEST_DIR. This implicit contract isn't obvious to callers. Rename the param or add a doc comment:
rust// images_dir: path relative to the crate root (CARGO_MANIFEST_DIR)
3. matched_fixture logic is misleading
The eprintln! for "no fixtures matching prefix" fires even when the directory exists but is intentionally empty (e.g., first-time setup). This could confuse contributors. A #[cfg(not(windows))] guard would eliminate the need for this runtime check entirely.
4. Cargo.toml dep split is correct but incomplete
Gating aprilgrid/apriltag under cfg(not(windows)) is the right call, but any test code that uses those crates also needs #[cfg(not(windows))] guards — otherwise the code won't compile on Windows even with the deps absent. The PR doesn't show whether that's handled in the test bodies.

**Comment by Sunilkumarchavhan9** [comment]
hi  @sidd-27  
Updated the PR based on your feedback.

Changes made:
- Replaced the runtime `eprintln!` + `Ok(())` skip behavior with explicit `#[cfg_attr(windows, ignore = "...")]` on the fixture-driven tests, so Windows now reports them as ignored instead of silently passing.
- Made the fixture path contract explicit in the helper.
- Removed the misleading runtime "no fixtures matching prefix" skip path and now fail loudly on non-Windows if expected fixtures are not present.
- Also completed the Windows gating for the benchmark targets so `cargo check -p kornia-apriltag --all-targets` succeeds on Windows.

Verified on Windows:
- `cargo test -p kornia-apriltag` -> 35 passed, 8 ignored, 0 failed
- `cargo check -p kornia-apriltag --all-targets` -> finished successfully

File: crates/kornia-apriltag/benches/bench_decoding.rs [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+#[cfg(not(windows))]
 use kornia_apriltag::{family::TagFamilyKind, AprilTagDecoder, DecodeTagsConfig};
+#[cfg(not(windows))]
 use kornia_image::{allocator::CpuAllocator, Image};
+#[cfg(not(windows))]
 use kornia_imgproc::color::gray_from_rgb_u8;
+#[cfg(not(windows))]
 use kornia_io::jpeg::read_image_jpeg_rgb8;
+#[cfg(not(windows))]
 use std::path::PathBuf;
```
Comment (sidd-27) [inline crates/kornia-apriltag/benches/bench_decoding.rs:14]: is there a cleaner way to get this done?
